### ISSUE TITLE
sbt --client fails on M1 Mac docker instance

### ISSUE THREAD

**Issue by wbillingsley** [sbt/sbt#6906]
## steps

On an m1 mac, in a docker instance, e.g. Devcontainer for https://github.com/theIntelligentBook/supercollaborative , 

sbt --client

## problem

sbtn attempts to load ld-linux-x86-64.so.2 library despite running on ARM architecture

```
root@4e6c90a02769:/workspaces/supercollaborative# sbt --client
downloading sbtn 1.4.7
qemu-x86_64: Could not open '/lib64/ld-linux-x86-64.so.2': No such file or directory
```

## expectation

sbt client runs and connects to (or starts) sbt server

## notes

Dockerfile uses openjdk:17-jdk-slim-bullseye as its base, with sbt installed via Coursier (`RUN bash -c 'cs install sbt'`)

Edit: sbt --java-client seems to work, but perhaps could be suggested from the error response or docs?

**Comment by mkurz** [comment]
I get someting similiar running sbt 1.7.0-M2 inside a Ubuntu 22.04 virtual machine on my Macbook Pro M1:
```sh
$ sbt --client
/home/mkurz/.sdkman/candidates/sbt/1.7.0-M2/sbt: line 222: /home/mkurz/.cache/sbt/boot/sbtn/1.4.7/sbtn: cannot execute binary file: Exec format error
/home/mkurz/.sdkman/candidates/sbt/1.7.0-M2/sbt: line 222: /home/mkurz/.cache/sbt/boot/sbtn/1.4.7/sbtn: Success
```

**Comment by eed3si9n** [comment]
I guess it's because we are currently assuming "Linux" to be Linux on x86_64, whereas you guys are running Linux on ARM?

**Comment by mkurz** [comment]
Yeah, that's why probably.

**Comment by mkurz** [comment]
With sbt 1.8 I get a better error message:
```sh
$ sbt --client
/home/mkurz/.sdkman/candidates/sbt/1.8.0/bin/sbt: line 222: /home/mkurz/.sdkman/candidates/sbt/1.8.0/bin/sbtn-x86_64-pc-linux: cannot execute binary file: Exec format error
/home/mkurz/.sdkman/candidates/sbt/1.8.0/bin/sbt: line 222: /home/mkurz/.sdkman/candidates/sbt/1.8.0/bin/sbtn-x86_64-pc-linux: Success
```

See here:
https://github.com/sbt/sbt/blob/09e06c45f01a72ed1010873a3ba89c0b70689fca/sbt#L174-L176

To solve this we need to generate and host a linux aarch64 binary in the sbtn-dist repo:
https://github.com/sbt/sbtn-dist/blob/develop/.github/workflows/native.yml
Looking at how the scala-cli binary for aarch64 gets build (https://github.com/VirtusLab/scala-cli/pull/1703), I think we can also make use of https://github.com/uraimo/run-on-arch-action.

**Comment by mkurz** [comment]
I build an aarch64 `sbtn` binary locally, but ran into following error: #7095

**Comment by mkurz** [comment]
I am pretty sure this can be closed now that we have
* #7110
* #7111
* #7112

so this should hopefully be fixed with next sbt 1.8.1 release.
